lib/rubocop/ast/processed_source.rb

```diff
@@ -165,13 +165,15 @@ def parse(source, ruby_version)
       def tokenize(parser)
         begin
           ast, comments, tokens = parser.tokenize(@buffer)
-
-          ast.respond_to?(:complete!) && ast.complete!
+          ast ||= nil # force `false` to `nil`, see https://github.com/whitequark/parser/pull/722
         rescue Parser::SyntaxError
           # All errors are in diagnostics. No need to handle exception.
+          comments = []
+          tokens = []
         end
 
-        tokens = tokens.map { |t| Token.from_parser_token(t) } if tokens
+        ast&.complete!
+        tokens = tokens.map { |t| Token.from_parser_token(t) }
```

Comment:
Is `tokens` an object shared with other code? Or could `tokens.map!` be a viable optimization here?

Reply:
Sure, I'll change it to `map!`

Reply:
Done 👍